File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/sale/model/entity/SaleDeliveryEntity.java
package com.ccadmin.app.sale.model.entity;

import com.ccadmin.app.sale.model.constants.SaleConstants;
import com.ccadmin.app.shared.model.entity.AuditTableEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

@Entity
@Table(name = "sale_delivery")
public class SaleDeliveryEntity extends AuditTableEntity implements Serializable {

    @Id
    public String SaleCod;
    public String DeliveryTypeCod;
    public String DeliveryStatus = SaleConstants.DELIVERY_STATUS_PENDING;
    public Long ClientAddressID;
    public String IsThirdParty = "N";
    public String Names;
    public String DocumentType;
    public String DocumentNumber;
    public String Phone;
    public String Email;
    public String Address;
    public String GeocodedAddress;
    public String Reference;
    public String CountryCod;
    public String CountryName;
    public String StateName;
    public String CityName;
    public String UbigeoCod;
    public BigDecimal Latitude;
    public BigDecimal Longitude;
    public String Instructions;
    public BigDecimal EstimatedDistanceKm;
    public Date ScheduledFrom;
    public Date ScheduledTo;
    public String ShippingProviderCod;
    public String TrackingNumber;
    public String AgencyName;
    public String AgencyAddress;
    public Date ReadyDate;
    public Date DispatchDate;
    public Date DeliveredDate;
    public String Commenter;

    public SaleDeliveryEntity() {
    }
}

File: frontend/web_ccadmin_lt/src/app/enterprise/sale/model/entity/SaleDeliveryEntity.ts
import { AuditTableEntity } from 'src/app/enterprise/shared/model/entity/AuditTableEntity';

export class SaleDeliveryEntity extends AuditTableEntity {
  public SaleCod: string = '';
  public DeliveryTypeCod: string = '';
  public DeliveryStatus: string = '';
  public ClientAddressID: number | null = null;
  public IsThirdParty: string = 'N';
  public Names: string = '';
  public DocumentType: string = '';
  public DocumentNumber: string = '';
  public Phone: string = '';
  public Email: string = '';
  public Address: string = '';
  public GeocodedAddress: string = '';
  public Reference: string = '';
  public CountryCod: string = '';
  public CountryName: string = '';
  public StateName: string = '';
  public CityName: string = '';
  public UbigeoCod: string = '';
  public Latitude: number | null = null;
  public Longitude: number | null = null;
  public Instructions: string = '';
  public EstimatedDistanceKm: number | null = null;
  public ScheduledFrom: Date | null = null;
  public ScheduledTo: Date | null = null;
  public ShippingProviderCod: string = '';
  public TrackingNumber: string = '';
  public AgencyName: string = '';
  public AgencyAddress: string = '';
  public ReadyDate: Date | null = null;
  public DispatchDate: Date | null = null;
  public DeliveredDate: Date | null = null;
  public Commenter: string = '';

  constructor() {
    super();
  }
}
